### PR TITLE
#28 - Landing Page Hero

### PR DESCRIPTION
**Pull request by jhancock532**
[Ticket on Codebase](https://projects.torchbox.com/projects/contentful-careers-page/tickets/28).

Added a landing page hero component as per the designs on Figma. The navigation is included in a separate ticket. In the future we'll need to add a rich text renderer component as a child of this component - see the todo comment left in `index.tsx`.

I've attempted to use Next/Image for this but the restrictive style rules applied to it make it very difficult to restyle. I've included a simple `img` in its place, although this will need updating with appropriate `srcset` values when I know what `srcset` Contentful offers.

Designs for tablet and mobile have been improvised.

## Screenshots
image
image
image
image

### CODE REVIEW

File: components/LandingPage/LandingPageHero.tsx [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+                />
+            </ShardClip>
+
+            <div className={styles.textContainer}>
+                <h1 className={styles.title}>
+                    <span className={styles.titleAccent}>Being at</span>
+                    <br />
+                    Torchbox
+                </h1>
+                <div className="subtitle">{children}</div>
```
Comment (jhancock532) [inline components/LandingPage/LandingPageHero.tsx:30]: "subtitle" should be {style.subtitle}

File: components/LandingPage/LandingPageHero.tsx [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+};
+
+export const LandingPageHero = ({ image, children }: LandingPageHeroProps) => {
+    return (
+        <>
+            <svg width="0" height="0" viewBox="0 0 1 1">
+                <defs>
+                    <clipPath
+                        id="shardClipPath"
+                        clipPathUnits="objectBoundingBox"
```
Comment (chris-lawton) [inline components/LandingPage/LandingPageHero.tsx:17]: question: what does this do?

File: components/LandingPage/LandingPageHero.tsx [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+
+type LandingPageHeroProps = {
+    image: ImageTypes;
+    children: React.ReactNode;
+};
+
+export const LandingPageHero = ({ image, children }: LandingPageHeroProps) => {
+    return (
+        <>
+            <svg width="0" height="0" viewBox="0 0 1 1">
```
Comment (chris-lawton) [inline components/LandingPage/LandingPageHero.tsx:13]: question: i'm curious why are the width and height set to 0?

File: styles/LandingPageHero.module.scss [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+            }
+
+            @include media-query(large) {
+                width: 575px !important;
+                height: 575px !important;
+            }
+
+            @include media-query(hero-break) {
+                width: 750px !important;
+                height: 775px !important;
```
Comment (chris-lawton) [inline styles/LandingPageHero.module.scss:113]: question: are all the above `!important`'s required?

Comment (chris-lawton) [reply styles/LandingPageHero.module.scss:113]: ah possibly to do with `Next/Image`

File: components/LandingPage/LandingPageHero.tsx (hunk 2 above)
Comment (jhancock532) [reply components/LandingPage/LandingPageHero.tsx:17]: This sets the clip path to be relative to the object that is being clipped, on a scale of 0 to 1. If we used the other setting, `userSpaceOnUse`, we'd be clipping according to the co-ord system of the svg viewbox of the `<div>`, which is non-existent. [MDN docs](https://developer.mozilla.org/en-US/docs/Web/SVG/Attribute/clipPathUnits).

File: components/LandingPage/LandingPageHero.tsx (hunk 3 above)
Comment (jhancock532) [reply components/LandingPage/LandingPageHero.tsx:13]: I did this initially to hide the SVG, but as the SVG doesn't contain any renderable content I think it is safe to remove these. There may be some cross browser issues with this however - I've double checked clipPathUnits and it works pretty much everywhere.

File: styles/LandingPageHero.module.scss (hunk 4 above)
Comment (jhancock532) [reply styles/LandingPageHero.module.scss:113]: Yep, these were added to override the default width and height of the image as Next/Image sets these internally twice over.

File: components/LandingPage/LandingPageHero.tsx (hunk 2 above)
Comment (jhancock532) [reply components/LandingPage/LandingPageHero.tsx:17]: I've tested this across browsers with Browserstack and removed the unnecessary viewBox and width & height definitions

**Pull request merged**